### PR TITLE
progress of innoextract extraction

### LINKED ISSUE #3874: Update Innoextract to compile with boost 1.85
**Describe the bug**
/run/media/disk_2/vcmi/vcmi/launcher/lib/innoextract/src/stream/slice.cpp:162: error: ‘directory_iterator’ is not a member of ‘boost::filesystem’; did you mean ‘directory_entry’?
/run/media/disk_2/vcmi/vcmi/launcher/lib/innoextract/src/stream/slice.cpp: In member function ‘bool stream::slice_reader::open_file_case_insensitive(const path_type&, const path_type&)’:
/run/media/disk_2/vcmi/vcmi/launcher/lib/innoextract/src/stream/slice.cpp:162:28: error: ‘directory_iterator’ is not a member of ‘boost::filesystem’; did you mean ‘directory_entry’?
  162 |         boost::filesystem::directory_iterator end;
      |                            ^~~~~~~~~~~~~~~~~~
      |                            directory_entry

**Additional context**
Fixed in (https://github.com/dscharrer/innoextract/commit/264c2fe6b84f90f6290c670e5f676660ec7b2387)

### PR DESCRIPTION
**Pull request by Laserlicht**
show process of innoextract extraction now.

image

also fixes #3874

### CODE REVIEW

File: launcher/firstLaunch/firstlaunch_moc.cpp [hunk 1]
```diff
@@ -341,7 +341,13 @@ void FirstLaunchView::extractGogData()
 
 			o.preserve_file_times = true; // also correctly closes file -> without it: on Windows the files are not written completly
 
-			process_file(tmpFileExe.toStdString(), o);
+			this->setEnabled(false);
+			process_file(tmpFileExe.toStdString(), o, [this](float percent) {
+				ui->pushButtonGogInstall->setText(tr("Installing...") + " " + QString::number(static_cast<int>(percent * 100)) + "%");
```
Comment (IvanSavenko) [inline launcher/firstLaunch/firstlaunch_moc.cpp:346]: Please check, not sure how to show '%' sign in Qt formatting. You might need '%%' instead
```suggestion
			process_file(tmpFileExe.toStdString(), o, [this](float progress) {
				ui->pushButtonGogInstall->setText(tr("Installing... %1%").arg(progress* 100));
```

Oh, and your variable is not 'percent', this is progress or ratio. Percentages should go from 0 to 100, but this parameter goes from 0 to 1, so should be named appropriately

**Comment by IvanSavenko** [comment]
Progress bar in place of button would have been perfect, but definitely better than nothing.
Was rather confused first time I used it due to lack of progress indication.

File: launcher/firstLaunch/firstlaunch_moc.cpp (hunk 1 above)
Comment (kambala-decapitator) [reply launcher/firstLaunch/firstlaunch_moc.cpp:346]: don't think you need %%

is extracting done on the main thread? I think it's better to add `qApp->processEvents()` to make sure that UI updates properly. If not on main thread, then you need to use `ui->pushButtonGogInstall->invoke()` iirc

File: launcher/firstLaunch/firstlaunch_moc.cpp [hunk 2]
```diff
@@ -341,7 +341,13 @@ void FirstLaunchView::extractGogData()
...
+				ui->pushButtonGogInstall->update();
```
Comment (kambala-decapitator) [inline launcher/firstLaunch/firstlaunch_moc.cpp:347]: this line shouldn't be needed

File: launcher/firstLaunch/firstlaunch_moc.cpp [hunk 3]
```diff
@@ -341,7 +341,13 @@ void FirstLaunchView::extractGogData()
...
+				qApp->processEvents();
+			});
+			this->setEnabled(true);
```
Comment (kambala-decapitator) [inline launcher/firstLaunch/firstlaunch_moc.cpp:350]:
```suggestion
			setEnabled(true);
```

**Pull request merged**